docs/OperatorBestPractices.md

```diff
@@ -0,0 +1,103 @@
+
+## Making an Operator Managed Backing Service Bindable
+
+In order to make a service bindable, the operator provider needs to express
+the information needed by applications to bind with the services provided by
+the operator. In other words, the operator provider must express the
+information that is “interesting” to applications.
+
+There are three methods for making Operator Managed Backing Service Bindable:
+
+* [Operator Providing Metadata in OLM](#operator-providing-metadata-in-olm)
+* [Operator Providing Metadata not in OLM](#operator-providing-metadata-not-in-olm)
+* [Operator Not Providing Metadata](#operator-not-providing-metadata)
+
+### Operator Providing Metadata in OLM
+
+This feature enables operator providers to specify binding information an
+operator's OLM (Operator Lifecycle Manager) descriptor. The Service Binding
+Operator extracts to bind the application together with the backing service.
+The information may be specified in the "status" and/or "spec" section of the
+OLM in plaintext or as a reference to a secret.
+
+For example, this is an example of the *bind-able* operator OLM Descriptor.
+In this case for a PostgreSQL database backing operator. 
+``` yaml
+---
+[...]
+statusDescriptors:
+  description: Name of the Secret to hold the DB user and password
+    displayName: DB Password Credentials
+    path: dbCredentials
+    x-descriptors:
+      - urn:alm:descriptor:io.kubernetes:Secret
+      - binding:env:object:secret:user
+      - binding:env:object:secret:password
+  description: Database connection IP address
+    displayName: DB IP address
+    path: dbConnectionIP
+    x-descriptors:
+      - binding:env:attribute
+```
+
+The operator is available [here.](https://github.com/operator-backing-service-samples/postgresql-operator)
+
+### Operator Providing Metadata not in OLM
+
+This feature enables operator prividers who do not use OLM (Operator Lifecycle
+Manager) to provide metadata outside of an OLM descriptor. In this method,
+the binding information is provided as annotations in the CRD of the operator
+that manages the backing service. The Service Binding Operator extracts the
+annotations to bind the application together with the backing service.
+
+For example: 
+``` yaml
+---
+[...]
+kind: CustomResourceDefinition
+apiVersion: apiextensions.k8s.io/v1beta1
+metadata:
+  name: databases.postgresql.baiju.dev
+  annotations:
+    servicebindingoperator.redhat.io/status.dbConfigMap.password: 'volume:configmap'
+    servicebindingoperator.redhat.io/status.dbConfigMap.username: 'env:configmap'
+    servicebindingoperator.redhat.io/status.dbName: env
+```
+
+### Operator Not Providing Metadata
+
+This feature enables operators that manage backing services but which don't
+have any metadata in their CSV to use the Service Binding Operator to bind
+together the service and applications. The Service Binding Operater binds all
+sub-resources defined in the backing service CR by populating the binding
```

Comment:
This is how a resource and sub-resource relationship is set. Let's mention that:
https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#owners-and-dependents

Reply:
Nice reference - done!